lib/companion_web/plugs/auth_plugs.ex

```diff
@@ -7,15 +7,31 @@ defmodule CompanionWeb.Plugs.RequireLoggedIn do
 
   def init(default), do: default
 
-  def call(conn, url) do
-    case get_session(conn, :user) do
-      nil ->
-        conn
-        |> redirect(to: url)
-        |> halt
+  def call(%{assigns: %{user: nil}} = conn, url) do
```

Comment:
Maybe rename `url` to `auth_url` here? I was a bit confused until I realised that it comes from configuration rather than the request destination.